rust_src/src/frames.rs

```diff
@@ -1,21 +1,110 @@
 //! Generic frame functions.
 
 use remacs_macros::lisp_fn;
-use remacs_sys::{selected_frame as current_frame, Lisp_Frame};
-
+use remacs_sys::{fget_output_method, fget_terminal, selected_frame as current_frame,
+                 Fselect_window, Lisp_Frame, OutputMethod, Qns, Qw32, Qx};
 use lisp::{ExternalPtr, LispObject};
+use objects::eq;
+use std::ptr;
 use lisp::defsubr;
 
 pub type LispFrameRef = ExternalPtr<Lisp_Frame>;
 
+impl LispFrameRef {
+    #[inline]
+    pub fn is_live(&self) -> bool {
+        if unsafe { fget_terminal(self.as_ptr()) == ptr::null() } {
```

Comment:
Why not write this as `fget_terminal != null()` and skip the if/else?

Reply:
Oops :)

Reply:
Or rather, `!fget_terminal.is_null()`